docs/DeveloperGuide.md

```diff
@@ -250,9 +252,27 @@ The `Person` class is extended by three other classes, each with their own addit
 
 #### Design Considerations
 
-* **Flexible Data Handling**: Fields not applicable to all roles are made optional within the `Person` class to enable a scalable and adaptable system architecture.
-
-* **Role-Specific UI Elements**: The decision to dynamically adjust the UI based on the person's role enhances the overall user experience by providing context-sensitive information.
+**Aspect: How to distinguish the person role:**
+
+* **Alternative 1 (current choice)**: Creation of subclasses `Supplier`, `Employee`, and `Client` that inherit from the `Person` class.
+   * **Pros**:
+       * Specialization: Allows for clear role-specific methods and properties, ensuring that objects have only the attributes and behaviors pertinent to their role.
+       * Extensibility: Easier to add new roles by creating additional subclasses.
+       * Polymorphism: Enables the use of a single interface to represent any person type, which simplifies code that interacts with these objects.
```

Comment:
Can we use ordered list for this to maintain consistency with the format in the `delete` implementation section? 😄 
![image](https://github.com/AY2324S2-CS2103T-F12-1/tp/assets/93519596/48a4955e-a726-45e7-81f5-bde23d7e2a2b)

Reply:
Thanks!